code/modules/mining/remote_fob.dm

```diff
@@ -0,0 +1,329 @@
+/////////For remote construction of FOB using a computer on the ship during setup phase
+GLOBAL_LIST_INIT(dropship_lzs, typecacheof(list(/area/shuttle/drop1/lz1, /area/shuttle/drop2/lz2)))
+
+
+/////////////////////////////// Drone Mob
+
+/mob/camera/aiEye/remote/fobdrone
+	name = "Remote Construction Drone"
+	mouse_opacity = MOUSE_OPACITY_OPAQUE
+	icon = 'icons/Marine/remotefob.dmi'
+	icon_state = "drone"
+	var/area/starting_area
+	use_static = FALSE
+	var/turf/spawnloc
+
+/mob/camera/aiEye/remote/fobdrone/Initialize()
+	. = ..()
+	starting_area = get_area(loc)
+
+/mob/camera/aiEye/remote/fobdrone/setLoc(atom/t) //unrestricted movement inside the landing zone
+	var/area/curr_area = get_area(t)
+	if(istype(t, /turf/closed) || locate(/obj/machinery/door/poddoor) in t)
+		return
+	if(curr_area == starting_area || is_type_in_typecache(curr_area, GLOB.dropship_lzs))
+		return ..()
+
+
+/mob/camera/aiEye/remote/fobdrone/relaymove(mob/user, direct)
+	dir = direct //This camera eye is visible as a drone, and needs to keep the dir updated
+	return ..()
+
+/mob/camera/aiEye/remote/fobdrone/update_remote_sight(mob/living/user)
+	user.see_invisible = FALSE
+	user.sight = SEE_SELF|SEE_BLACKNESS
+	user.lighting_alpha = LIGHTING_PLANE_ALPHA_NV_TRAIT
+	user.see_in_dark = 7
+	return TRUE
+
+/////////////////////////////// the camera computer
+
+/obj/machinery/computer/camera_advanced/remote_fob
+	name = "FOB Construction Drone Control"
+	desc = "A computer console equipped with camera screen and controls for a planetside deployed construction drone. Materials or equipment vouchers can be added simply by inserting them into the computer."
+	icon_state = "syndishuttle"
+	req_access = list(ACCESS_MARINE_REMOTEBUILD)
+	networks = FALSE
+	off_action = new/datum/action/innate/camera_off/remote_fob
+	jump_action = null
+	var/drone_creation_allowed = TRUE
+	var/obj/docking_port/stationary/marine_dropship/spawn_spot
+	var/lz
+	var/datum/action/innate/remote_fob/metal_cade/metal_cade = new
```

Comment:
Assigning actions on the definition is generally necessary for a `static` type or the likes.
Not being the case, prefer doing it on `Initialize()` as that gives the MC finer control to improve the startup times.

Reply:
wont compile if i stuff them in initialize

Reply:
This is awful, do it properly in initialize

Reply:
How are you doing it on initialize? The correct way would be something like this:

```
/obj/machinery/computer/camera_advanced/remote_fob
	var/datum/action/innate/remote_fob/metal_cade/metal_cade

/obj/machinery/computer/camera_advanced/remote_fob/Initialize()
	. = ..()
	metal_cade = new()
```